h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoderMojoWriter.java

```diff
@@ -0,0 +1,48 @@
+package ai.h2o.automl.targetencoding;
+
+import hex.Model;
+import hex.ModelMojoWriter;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class TargetEncoderMojoWriter extends ModelMojoWriter {
+
+  public TargetEncoderMojoWriter(Model model) {
+    super(model);
+  }
+
+  @Override
+  public String mojoVersion() {
+    return "1.00";
+  }
+
+  @Override
+  protected void writeModelData() throws IOException {
+    writeTargetEncodingMap();
+  }
+
+  @Override
+  protected void writeExtraInfo() throws IOException {
+    // Do nothing
+  }
+
+  /**
+   * Writes encoding map into the file line by line
+   */
+  private void writeTargetEncodingMap() throws IOException {
+    Map<String, Map<String, int[]>> targetEncodingMap = ((TargetEncoderModel) model)._output._target_encoding_map;
+    if(targetEncodingMap != null) {
+      startWritingTextFile("feature_engineering/target_encoding.ini");
```

Comment:
What is the reason for using ini format? Can't we use something more generic like JSON?

Reply:
@mn-mikke This is how all Mojo writers are implemented right now. Why JSON was not chosen in a first place.... we should ask designers. Maybe the goal was to reduce dependence on third party libraries( with JSON you would most likely need one). Also JSON is shining in hierarchical data structures and maybe here we really just needed a config-like format.  In Scala JSON is I would say a first class citizen because it is so easy to create hierarchies with value classes `case class`... in Java it is less appealing.

Reply:
@deil87 JSON is already used in MOJO representation. See `ModelMojoWriter.writeModelDetails()`

Reply:
@mn-mikke Ok, I was not careful in saying `all writers`.  But anyway.... there are comments that it is kind of `experimental`.  I don't know why they experimented with this. I would say that current way of writing to the file is more efficient in terms of space than JSON. And I did not reinvent the wheel, just reused existing methods. What are the objectives of your comment? What benefits for a particular scenario(writing encoding maps) it will bring?

Reply:
@mn-mikke @deil87 keep in mind the writeModelDetails is not actually used in the scoring path, this is (mostly) for MOJO debugging. 

I think using current structures makes sense unless there is a good reason to do so, I wouldn't use JSON to represent a dictionary myself. We need to keep in mind that TE structures might be large. JSON is very verbose and adds tons of overhead. Genmodel also depends on opencsv, that would be more appropriate than JSON.

Reply:
@michalkurka do you suggest to switch to the opencsv approach? If yes, can you point out to an example in the code where it is being used? But IMHO for a current task writing in ini format is fine.